packages/core/src/QueryApi.ts

```diff
@@ -40,6 +53,22 @@ export default interface QueryApi {
    */
   with(options: Partial<QueryOptions>): QueryApi
 
+  /**
+   * Creates a cold observable of the lines returned by the given query.
+   *
+   * @param query the query text in the format specifed in `QueryOptions.type`
+   * @param variables variables to add to the query, when `QueryOptions.type` is "flux"
+   */
+  lines(query: string, variables?: Variables): Observable<string>
```

Comment:
i didnt do this yet, because i thought the order of parameters felt strange then:
`queryLines(query, consumer, variables)`

Reply:
Yes, the callback should be the last by convention. It can be realized in a backward-compatible way.

*interface*:
```js
  queryLines(query: string, consumer: CommunicationObserver<string>): void
  queryLines(
    query: string,
    parameters: QueryParameters,
    consumer: CommunicationObserver<string>
  ): void
```
*implementation*:
```
  queryLines(
    query: string,
    params: any | CommunicationObserver<string>,
    consumer?: CommunicationObserver<string>
  ): void {
    if (arguments.length === 2) {
      consumer = params as CommunicationObserver<string>
    }
...
// consumer as CommunicationObserver<string> 
```